crates/common/tedge_config_macros/impl/src/query.rs

```diff
@@ -346,8 +435,31 @@ fn generate_string_writers(paths: &[VecDeque<&FieldOrGroup>]) -> TokenStream {
     ) = paths
         .iter()
         .zip(variant_names)
-        .map(|(path, variant_name)| {
-            let segments = path.iter().map(|thing| thing.ident()).collect::<Vec<_>>();
+        .map(|(path, (_enum_variant, _iter_variant, match_variant))| {
+            let mut id_gen = NumericIdGenerator::default();
+            let read_segments = path.iter().map(|&thing| {
+                let ident = thing.ident();
+                match thing {
+                    FieldOrGroup::Field(_) => quote!(#ident),
+                    FieldOrGroup::Group(_) => quote!(#ident),
+                    FieldOrGroup::Multi(_) => {
+                        let field = id_gen.next_id(ident.span());
+                        quote_spanned!(ident.span()=> #ident.get(#field.as_deref())?)
+                    },
+                }
+            }).collect::<Vec<_>>();
+            let mut id_gen = NumericIdGenerator::default();
+            let write_segments = path.iter().map(|&thing| {
+                let ident = thing.ident();
+                match thing {
+                    FieldOrGroup::Field(_) => quote!(#ident),
+                    FieldOrGroup::Group(_) => quote!(#ident),
+                    FieldOrGroup::Multi(_) => {
+                        let field = id_gen.next_id(ident.span());
+                        quote_spanned!(ident.span()=> #ident.get_mut(#field.as_deref())?)
```

Comment:
This `get_mut` is the only difference between the two cases. This duplication should be removed